src/features/semantic_tokens.zig

```diff
@@ -1088,17 +1100,111 @@ fn writeIdentifier(builder: *Builder, name_token: Ast.TokenIndex) error{OutOfMem
         tree.tokenStart(name_token),
     )) |child| {
         const is_param = child.decl == .function_parameter;
-
+        const mutable = !child.isConst();
         if (try child.resolveType(builder.analyser)) |decl_type| {
-            return try colorIdentifierBasedOnType(builder, decl_type, name_token, is_param, .{ .mutable = !child.isConst() });
+            return try colorIdentifierBasedOnType(
+                builder,
+                decl_type,
+                name_token,
+                is_param,
+                .{
+                    .static = !decl_type.isTypeOrFunc() and try child.isStatic(),
+                    .mutable = mutable,
+                },
+            );
         } else {
-            try writeTokenMod(builder, name_token, if (is_param) .parameter else .variable, .{ .mutable = !child.isConst() });
+            try writeTokenMod(
+                builder,
+                name_token,
+                if (is_param) .parameter else .variable,
+                .{
+                    .static = try child.isStatic(),
+                    .mutable = mutable,
+                },
+            );
         }
     } else {
         try writeToken(builder, name_token, .variable);
     }
 }
 
+fn writeFieldAccess(builder: *Builder, node: Ast.Node.Index) error{OutOfMemory}!void {
+    const handle = builder.handle;
+    const tree = builder.handle.tree;
+    const lhs_node, const field_name_token = tree.nodeData(node).node_and_token;
+
+    const symbol_name = offsets.identifierTokenToNameSlice(tree, field_name_token);
+
+    try writeNodeTokens(builder, lhs_node);
+
+    const lhs = try builder.analyser.resolveTypeOfNode(.{ .node = lhs_node, .handle = handle }) orelse {
+        const is_static = static_blk: {
+            const lhs_token = tree.nodeMainToken(lhs_node);
+            const lhs_decl = try builder.analyser.lookupSymbolGlobal(
+                handle,
+                tree.tokenSlice(lhs_token),
+                tree.tokenStart(lhs_token),
+            ) orelse break :static_blk false;
+            break :static_blk try lhs_decl.isStatic();
+        };
+        try writeTokenMod(
+            builder,
+            field_name_token,
+            .variable,
+            .{ .static = is_static },
+        );
```

Comment:
This code assumes that the lhs of a field access is guaranteed to be an identifier even though it could be any kind of expression.

This still passes all the tests:
```suggestion
        try writeToken(builder, field_name_token, .variable);
```
No need to over complicate this.